pkgs/desktops/cinnamon/xapps/default.nix

```diff
@@ -0,0 +1,50 @@
+{ fetchFromGitHub, glib, gobject-introspection, gtk3, libgnomekbd, libxklavier, meson, ninja, pkgconfig, python3, python3Packages, stdenv, vala, wrapGAppsHook }:
+
+stdenv.mkDerivation rec {
+  pname = "xapps";
+  version = "1.6.4";
+  enableParallelBuilding = true;
+
+  src = fetchFromGitHub {
+    owner = "linuxmint";
+    repo = pname;
+    rev = version;
+    sha256 = "08c8j7wv56zf8mfgjfhd0wc1bbrw0dj1lmlawvsxwfajnnyjsm0d";
+  };
+
+  NIX_CFLAGS_COMPILE = [ "-I${glib.dev}/include/gio-unix-2.0" ]; # TODO: https://github.com/NixOS/nixpkgs/issues/36468
+
+  patches = [
+    ./py-override.patch # details see https://github.com/linuxmint/xapps/pull/65
+  ];
+
+  buildInputs = [ glib gtk3 libgnomekbd libxklavier pkgconfig python3Packages.pygobject3 vala ];
+  nativeBuildInputs = [ meson gobject-introspection ninja python3 wrapGAppsHook ];
+
+  mesonFlags = [
+    "-Dpy-overrides-dir=${placeholder "out"}/${python3.sitePackages}/gi/overrides"
+  ];
+
+  preBuild = ''
+    chmod +x /build/source/libxapp/g-codegen.py
+    patchShebangs /build/source/libxapp/g-codegen.py
+    pwd
+    '';
```

Comment:
why patch this after the fact? (could be done in postPatch?)
* https://github.com/linuxmint/xapps/blob/31133349e33260659e76c83f6edae57c38b0dc2b/libxapp/g-codegen.py